clinvar/conflict_huntr.py

```diff
@@ -0,0 +1,545 @@
+    review_status: str
+
+
+def chunks(iterable, chunk_size):
+    """
+    Yield successive n-sized chunks from an iterable
+
+    Args:
+        iterable (): any iterable - tuple, str, list, set
+        chunk_size (): size of intervals to return
+
+    Returns:
+        intervals of requested size across the collection
+    """
+
+    if isinstance(iterable, set):
+        iterable = list(iterable)
+
+    for i in range(0, len(iterable), chunk_size):
+        yield iterable[i : (i + chunk_size)]
+
+
+def generator_chunks(generator, size):
+    """
+    Iterates across a generator, returning specifically sized chunks
+
+    Args:
+        generator (): any generator or method implementing yield
+        size (): size of iterator to return
+
+    Returns:
+        a subset of the generator results
+    """
+    iterator = iter(generator)
+    for first in iterator:
+        yield list(chain([first], islice(iterator, size - 1)))
+
+
+def get_allele_locus_map(summary_file: str) -> tuple[dict, set]:
+    """
+    Process variant_summary.txt
+     - links the allele ID, Locus/Alleles, and variant ID
+    relevant fields:
+    0 #AlleleID
+    20 Chromosome
+    30 VariationID
+    31 Start
+    32 ReferenceAllele
+    33 AlternateAllele
+
+    Args:
+        summary_file (str): path to the gzipped text file
+
+    Returns:
+
+    """
+
+    allele_dict = {}
+    all_contigs = set()
+
+    for line in lines_from_gzip(summary_file):
+        if 'GRCh37' in line:
+            continue
+
+        # pull values from the line
+        allele_id = line[0]
+        chromosome = line[18] if 'chr' in line[18] else f'chr{line[18]}'
+        var_id = line[30]
+        pos = int(line[31])
+        ref = line[32]
+        alt = line[33]
+
+        # skip chromosomal deletions and insertions, mito, or massive indels
+        # this might break hail?
+        if (
+            ref == 'na'
+            or alt == 'na'
+            or 'm' in chromosome.lower()
+            or (len(ref) + len(alt)) > LARGEST_COMPLEX_INDELS
+        ):
+            continue
+
+        all_contigs.add(chromosome)
+
+        allele_dict[var_id] = {
+            'allele': allele_id,
+            'chrom': chromosome,
+            'pos': pos,
+            'ref': ref,
+            'alt': alt,
+        }
+
+    return allele_dict, all_contigs
+
+
+def lines_from_gzip(filename: str) -> str:
+    """
+    generator for gzip reading
+    copies file to local prior to reading
+
+    Args:
+        filename (): the gzipped input file
+
+    Returns:
+        generator - yields each line
+    """
+
+    if isinstance(to_path(filename), CloudPath):
+        tempfile = 'file.txt.gz'
+        to_path(filename).copy(tempfile)
+        filename = tempfile
+
+    with gzip.open(filename, 'rt') as handle:
+        for line in handle:
+            if line.startswith('#'):
+                continue
+            yield line.rstrip().split('\t')
+
+
+def county_county(subs: list[Submission]) -> Consequence:
```

Comment:
Um, is this a thing?

Reply:
Hah! good spot. I changed this to a better name in the parallel PR, but didn't bring it over